app/models/plan.rb

```diff
@@ -93,13 +93,10 @@ def visible
     plan.has_many :plan_metrics, foreign_key: :plan_id, &Logic::MetricVisibility::OfMetricAssociationProxy
   end
 
-  has_many :features_plans, :as => :plan
-
-  # FIXME: this should be a simple HABTM
-  # No it can't, because it is POLYMORPHIC
-  has_many :features, :through => :features_plans do
-    # returns all features owned by issuer, not only enabled by plan
-    def of_service
+  has_many :features_plans, as: :plan, dependent: :delete_all
```

Comment:
why did you add `dependent: :delete_all`?

Reply:
I don't remember, but it was a good reason, for sure!

It makes sense, though. `features_plans` is a N-N table between `features` and `plans`, and every record represents a feature enabled for a plan. It makes no sense to keep a feature enabled for a plan after the plan is removed.

Reply:
how many of them can there be. We split deleting for transactions failing reason when they are too big. That's why we have the background deletion.

Reply:
The plan having more features has 88. I don't think that's usual though, probably most of them have less than 10. Also, the association between `features` and `features_plans` is marked as `dependent: :delete_all` as well:

https://github.com/3scale/porta/blob/908be5de1dc76152c12f3d453c075defc2c30be9/app/models/feature.rb#L17